tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -0,0 +1,185 @@
+import { Box, Button, StyleableSection, Icon, Stack } from '../components';
+import { Window } from '../layouts';
+import { Color } from 'common/color';
+import { SFC } from 'inferno';
+import { JobToIcon } from './common/JobToIcon';
+import { deepMerge } from 'common/collections';
+
+type Job = {
+  unavailable_reason: string | null;
+  command: boolean;
+  open_slots: number;
+  used_slots: number;
+  icon: string;
+  prioritized: boolean;
+  description: string;
+};
+
+type Department = {
+  color: string;
+  jobs: Record<string, Job>;
+  open_slots: number;
+};
+
+type Data = {
+  departments_static: Record<string, Department>;
+  departments: Record<string, Department>;
+  alert_state: string;
+  shuttle_status: string;
+  disable_jobs_for_non_observers: boolean;
+  priority: boolean;
+  round_duration: string;
+};
+
+export const JobEntry: SFC<{
+  jobName: string;
+  job: Job;
+  department: Department;
+  onClick: () => void;
+}> = (data) => {
+  const jobName = data.jobName;
+  const job = data.job;
+  const department = data.department;
+  const jobIcon = job.icon || JobToIcon[jobName] || null;
+  return (
+    <Button
+      fluid
+      style={{
+        // Try not to think too hard about this one.
+        'background-color': job.unavailable_reason
+          ? '#949494' // Grey background
+          : job.prioritized
+            ? '#16fc0f' // Bright green background
+            : Color.fromHex(department.color)
+              .darken(10)
+              .toString(),
+        'color': job.unavailable_reason
+          ? '#616161' // Dark grey font
+          : Color.fromHex(department.color)
+            .darken(90)
+            .toString(),
+        'font-size': '1.1rem',
+        'cursor': job.unavailable_reason ? 'initial' : 'pointer',
+      }}
+      tooltip={
+        job.unavailable_reason ||
+        (job.prioritized ? (
+          <>
+            <p style={{ 'margin-top': '0px' }}>
+              <b>The HoP wants more people in this job!</b>
+            </p>
+            {job.description}
+          </>
+        ) : (
+          job.description
+        ))
+      }
+      onClick={() => {
+        !job.unavailable_reason && data.onClick();
+      }}>
+      <>
+        {jobIcon && <Icon name={jobIcon} />}
+        {job.command ? <b>{jobName}</b> : jobName}
+        <span
+          style={{
+            'white-space': 'nowrap',
+            'position': 'absolute',
+            'right': '0.5em',
+          }}>
+          {job.used_slots} / {job.open_slots}
+        </span>
+      </>
+    </Button>
+  );
+};
+
+export const JobSelection = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  if (!data?.departments_static) {
+    return null; // Stop TGUI whitescreens with TGUI-dev!
+  }
+  const departments: Record<string, Department> = deepMerge(
+    data.departments,
+    data.departments_static
+  ); // Why the fuck is it so hard to clone objects properly in JS?!
```

Comment:
I don't think you should be doing this at all but I won't block it

Reply:
And comment is kind of irrelevant now, so at least delete that.

Reply:
Treating them separately is honestly gonna cause more spaghetti and headaches than it's worth on my end. This is client-side JS, working on a relatively tiny amount of data, so performance isn't an issue.